### PR TITLE
webapps submit lazy accumulation

### PR DESCRIPTION
**Pull request by knguyen142**
## Summary

https://dimagi-dev.atlassian.net/browse/SAAS-11267
https://github.com/dimagi/commcare-hq/pull/28770

Sorry for the delay, finally got around to this and it turned out to be pretty straightforward. 

We were running into an issue where in slow internet connections the follow might happen:

1. user inputs an answer
2. webapps sends up an `answer` request
3. user clicks `submit` and webapps accumulates the answers in the form
4. webapps receives the `answer` response which updates the form with another question
5. webapps already has accumulated the incomplete answers and the submit request submits a incomplete form

incomplete-form

Solution:

Delays the answers accumulation until the submit request is picked up from the task queue.

1. user inputs an answer
2. webapps sends up an `answer` request
3.  user clicks `submit`
4. webapps receives the `answer` response which updates the form with another question
5. webapps accumulates the answers in the form which accumulates all the answers and submits all the answers in the form

webapps-submit-lazy

## Product Description

I'll leave this on staging for a while. Will create a QA ticket.

## Safety Assurance

- [x ] Risk label is set correctly
- [x ] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x ] If QA is part of the safety story, the "Awaiting QA" label is used
- [x ] I am certain that this PR will not introduce a regression for the reasons below

Will run through QA

### Safety story

Tested locally with simulated slow internet connection. Will leave on staging for a bit with QA.

### Rollback instructions

- [x ] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js
Comment (knguyen142) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js:91]: Following function arguments in https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting

**Review by orangejenny — APPROVED** [review]
Looks good. One thing I might have considered is making `requestParams` expect a function that returns an object, instead of adding another parameter to `serverRequest`. What do you think?

**Comment by knguyen142** [comment]
@orangejenny Good idea, I've reverted the changes and tried your approach. Let me know how it looks.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js
Comment (knguyen142) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js:465]: I think adding a attribute to a function is javascript pattern okay?

**Comment by knguyen142** [comment]
@orangejenny This has been [QA'd ](https://dimagi-dev.atlassian.net/browse/QA-2179)and I made the changes you suggested. While waiting for your check again, I'll merge this and create another PR if there are any changes (I believe those will be more stylistic than semantic, the functionality should be similar to the first approach).

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js:465]: Yeah, I think this is odd but okay. The only alternative I can think of is passing `action` as a separate parameter instead of as part of `requestParams`, then adding it to `requestParams` inside of `serverRequest`/`_serverRequest`.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
@knguyen142 Apologies for the delayed review, this looks reasonable to me.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js
Comment (knguyen142) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/webformsession.js:465]: This is actually what I did originally before switching to this, but it involved changing all of the things that called `serverRequest` and it felt very clunky.

**Pull request merged**
